### PR TITLE
chore: update remark and plugins

### PR DESCRIPTION
**Pull request by Trott**
Update to make this remark@13 compatible.

BREAKING CHANGE: remark@13 is a major release

### CODE REVIEW

**Comment by Trott** [comment]
This is likely blocked until https://github.com/nodejs/node/pull/35934 lands.

**Comment by nschonni** [comment]
Think I had a few others in https://github.com/nodejs/remark-preset-lint-node/pull/159, but I'm OK going with this one since I didn't resolved the conflicts over there. I'm still not sure how to handle the gfm plug-in though

**Comment by nschonni** [comment]
😆 looks like you also probably duplicated https://github.com/nodejs/node/pull/35647

**Comment by Trott** [comment]
Landed nodejs/node#35647 and re-running CI...

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,2466 @@
 {
   "name": "remark-preset-lint-node",
   "version": "1.17.1",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (nschonni) [inline package-lock.json:4]: Is this from a newer NPM version?

Comment (Trott) [reply package-lock.json:4]: Yes, npm@7.

**Comment by Trott** [comment]
Landed https://github.com/nodejs/remark-preset-lint-node/pull/159 instead.

**Pull request closed without merging**